### PR TITLE
reading the configuration file for mention to hostname

### PR DESCRIPTION
**Pull request by anthonyray**

### CODE REVIEW

File: action-sonos.py [hunk 1]
```diff
@@ -46,6 +43,11 @@
 else:
     logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
 
+# Connection
+HOSTNAME = CONFIGURATION['global']['hostname'] if CONFIGURATION['global']['hostname'] else "localhost"
+HERMES_HOST = "{}:1883".format(HOSTNAME)
+MOPIDY_HOST = HOSTNAME
```
Comment (paufabregat) [inline action-sonos.py]: No need for `MOPIDY_HOST`? Or am I missing something?

File: action-sonos.py [hunk 2]
```diff
@@ -46,6 +43,11 @@
...
+
```
Comment (paufabregat) [inline action-sonos.py]: Extra unnecessary blank line

**Review by paufabregat — CHANGES_REQUESTED** [review]
Please do same change you did for `action-sonos.py` for the following files that also need it: `action-custom.py`, `.../node/device_transport_control.py`, `.../node/music_playback_service.py` and `.../node/device_discovery_service.py`

**Comment by anthonyray** [comment]
@paufabregat I fixed the issues you pointed out !

**Comment by paufabregat** [comment]
@anthonyray you missed what I wrote here https://github.com/snipsco/snips-skill-sonos/pull/41#pullrequestreview-133708190

**Comment by anthonyray** [comment]
@paufabregat Here is the new approach. Tell me what you think !

File: snipssonos/services/service.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+from snipssonos.helpers.snips_config_parser import read_configuration_file
+
+
+class Service(object):
+    CONFIG = "config.ini"
+    CONFIGURATION = read_configuration_file(CONFIG)
```
Comment (paufabregat) [inline snipssonos/services/service.py]: I think we could add an improvement to lower code duplication. Here you could define a dummy HOST and PORT and a `get_port_from_config` and `get_host_from_config` methods. The methods would be called with the service name to pick up info from the config. 
You would have this in this class: 
~~~python
def get_port_from_config(service_name):
      self.PORT = self.CONFIGURATION['global'][service_name] if (
          self.CONFIGURATION['global'][service_name]) else self.PORT
~~~
and for instance in /music/search_service.py you could have: 
~~~python
self.PORT = self.get_port_from_config('music_service_port')
~~~
WDYT?

**Review by paufabregat — APPROVED** [review]
I like the idea! I think we could introduce an improvement to it, check my comment

File: snipssonos/services/service.py (hunk 3 above)
Comment (anthonyray) [reply snipssonos/services/service.py]: That's a good idea !

Comment (anthonyray) [reply snipssonos/services/service.py]: Actually, there's a huge huge side effect in my implementation. We are basically reading from the file system every time we create an instance of any service .... 
This is bad, I will find something better.

**Comment by anthonyray** [comment]
So I noticed that my approach introduced a big side effect : we are reading the file system everytime an instance of any service is created. This is not ok. I will fallback to another solution.

**Comment by anthonyray** [comment]
Check out my new approach ! We read only once to the filesystem and we pass around a configuration object to subclasses of Services.

**Review by paufabregat — APPROVED** [review]

**Pull request merged**